### PR TITLE
Update F# to .Net Core 2

### PR DESCRIPTION
**Pull request by adamchester**

### CODE REVIEW

File: fsharp-kestrel-sample/src/Program.fs [hunk 1]
```diff
@@ -1,55 +1,45 @@
...
-            .UseUrls("http://0.0.0.0:5001")
-            .UseStartup<Startup>()
-            .Build()
-            .Run()
-        
-        0 // return an integer exit code
\ No newline at end of file
+    let main args =
+
+        Log.Logger <- LoggerConfiguration()
```
Comment (bartelink) [inline fsharp-kestrel-sample/src/Program.fs:28]: maybe `use` the Logger and assign to `Log.Logger` after ?

File: fsharp-kestrel-sample/src/Program.fs [hunk 2]
```diff
@@ -1,55 +1,45 @@
...
+
+        Log.Information("Starting with arguments {Args}", args)
+
+        try
+            try
+                CreateWebHostBuilder(args).Build().Run()
+                successExitCode
+            with
+            | ex ->
+                Log.Fatal(ex, "Host terminated unexpectedly");
```
Comment (bartelink) [inline fsharp-kestrel-sample/src/Program.fs]: your C# is showing ;)

File: fsharp-kestrel-sample/src/Program.fs (hunk 1 above)
Comment (adamchester) [reply fsharp-kestrel-sample/src/Program.fs:28]: The goal was to be basically the same [the C# recommended pattern](https://github.com/serilog/serilog-docker/blob/master/web-sample/src/Program.cs#L33), unless there’s a great reason to deviate that I have considered yet?

File: fsharp-kestrel-sample/src/Program.fs (hunk 2 above)
Comment (adamchester) [reply fsharp-kestrel-sample/src/Program.fs]: Yes, I was trying to keep it similar, sorry ;)

Comment (bartelink) [reply fsharp-kestrel-sample/src/Program.fs]: `;`

Comment (adamchester) [reply fsharp-kestrel-sample/src/Program.fs]: Ahh! Good catch, thanks 👍

File: fsharp-kestrel-sample/src/Program.fs (hunk 1 above)
Comment (bartelink) [reply fsharp-kestrel-sample/src/Program.fs:28]: A try-catch-finally sure looks nice ;) I guess as general advice, assigning to `Log.Logger` and then doing `Log.CloseAndFlush` does make the most sense. (My comment as in reaction to all the nesting - normally I put one liner `with` or `finally` things on one line and/or go out of my way to use `use`, but writing it as clearly as possible including the nesting is probably no harm)

**Comment by adamchester** [comment]
OK I think this is ready.

File: fsharp-kestrel-sample/src/Program.fs (hunk 1 above)
Comment (adamchester) [reply fsharp-kestrel-sample/src/Program.fs:28]: Thanks @bartelink

**Comment by nblumhardt** [comment]
:+1:

**Pull request merged**